lib/knapsack_pro/report.rb

```diff
@@ -30,12 +30,19 @@ def self.save_subset_queue_to_file
       end
     end
 
-    def self.save_node_queue_to_api
+    def self.get_test_files
       test_files = []
       Dir.glob("#{queue_path}/*.json").each do |file|
         report = JSON.parse(File.read(file))
         test_files += report
       end
+      test_files
+    end
+
+    def self.save_node_queue_to_api(test_files = [])
```

Comment:
When `test_files` is not an empty array (when you provide test files that were recorded with the new formatter (`KnapsackPro::Formatters::TimeTracker`)).
then you may lose information about all test files fetched from Queue API if RSpec can't run a test file (i.e. it's pending), hence the missing 0s execution time for `spec/channels/test_channel_spec.rb` in the KS user dashboard.
Problem described here: https://github.com/KnapsackPro/knapsack_pro-ruby/pull/229#issuecomment-1814750491

All test files fetched from Queue API are saved with default execution time (0s) here:
https://github.com/KnapsackPro/knapsack_pro-ruby/blob/b0505b740c3f387670a50463375cb425cc9380d2/lib/knapsack_pro/tracker.rb#L63
Later on the 0s execution time is overridden by measured time. This way we don't lose info about tests assigned to CI node even when test runner (RSpec) won't run it.

Reply:
This has been resolved.